### PR TITLE
Request for code review

### PR DESCRIPTION
**Pull request by AymenBida**
### Members only

In this project we did : 

- Create users and posts models and controllers
- Use devise to authenticate users
- Create a feed page showing the posts and who made them if the user is logged in

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        <div class="collapse navbar-collapse" id="navbarSupportedContent">
+          <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+            <li class="nav-item">
+              <%= link_to 'Home', root_path, class: 'nav-link active', 'aria-current': 'page' %>
+            </li>
+            <% if user_signed_in? %>
+              <li class="nav-item">
+                <%= link_to 'Create a secret', new_post_path, class: 'nav-link text-white' %>
+              </li>
+            <% end %>
```
Comment (elmejdki) [inline app/views/layouts/application.html.erb]: Please remove logic from here and put it in view helpers

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          </ul>
+          <% if user_signed_in? %>
+            <span class="ml-auto nav-link text-white"><%= current_user.name %></span>
+            <%= link_to 'Sign Out', destroy_user_session_path, method: :delete, class: 'btn btn-outline-danger' %>
+          <% else %>
+            <%= link_to 'Log in', new_user_session_path, class: 'btn btn-outline-success ml-auto' %>
+            <%= link_to 'Sign up', new_user_registration_path, class: 'ml-2 btn btn-outline-light' %>
+          <% end %>
```
Comment (elmejdki) [inline app/views/layouts/application.html.erb]: Please remove logic from here and put it in view helpers

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <% if notice %>
+        <div class="alert alert-info" role="alert">
+          <%= notice %>
+        </div>
+      <% end %>
+      <% if alert %>
+        <div class="alert alert-danger" role="alert">
+          <%= alert %>
+        </div>
+      <% end %>
```
Comment (elmejdki) [inline app/views/layouts/application.html.erb]: Please remove logic from here and put it in view helpers

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  <% @posts.each do |post| %>
+  <div class="border rounded mb-2 p-4 mx-auto">
+  <div class="d-flex align-items-baseline">
+    <h2>
+      <%= post.title %>
+    </h2>
+    <% if user_signed_in? %>
+      <p class="text-secondary ml-2"> by <%= post.user.name %></p>
+    <% end %>
```
Comment (elmejdki) [inline app/views/posts/index.html.erb]: Please remove logic from here and put it in view helpers

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 

Great job guys :clap: :100: :+1:, you were almost gonna meet all the requirements for this project, but you still have some minor changes :hammer: :wrench: :hammer_and_wrench:  to take care of.
Please take the comments below:
:point_down: :point_down: :point_down: :point_down: 

## Required Changes :red_circle: 

##### PostsController

* [x] it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

##### Views & Helpers

* [x] View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)
* [x] There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

### Please make all the required changes, before you submit another code review.
## Happy Coding :tophat: 
_If you have any questions or you need help don't hesitate to contact me on slack._
## _Zakariae El Mejdki_

**Comment by mimipeshy** [comment]
### STATUS :APPROVED :partying_face: 

Hey there @AymenBida and @AshakaE   :wave: :wave:  good job on working on the requested changes. This project is therefore approved. Go ahead and merge. All the best in the next milestones :man_technologist: 

Regards
Peris Ndanu

**Pull request closed without merging**